### ISSUE TITLE
Circuit ORAM recursion broken by changes in Obliv-C and misc

### ISSUE THREAD

**Issue by samee** [samee/sqrtOram#1]
The current HEAD is broken on even circuit ORAM tests:

```bash
$ make build/test/testOramAccess
$ build/test/testOramAccess -- 1234 ckt --auto=512 &
$ build/test/testOramAccess localhost 1234 ckt --auto=512
[...stacktrace...]
```
The issue also exists with `bench/search`. Rebuilding with `make CFLAGS=-g` allows valgrind to look into the problem. `bench/search` was more time-tractable with valgrind:

```bash
$ make CFLAGS=-g build/bench/search
$ valgrind build/bench/search bench -t ckt --size=257 --axcount=6 :1234 &
$ valgrind build/bench/search bench -t ckt localhost:1234
```
[Logs attached](https://github.com/samee/sqrtOram/files/1182098/logs.txt)

Xiao noted in email that the code is significantly different from the one in published version. While we chase down regressions, I'll try to upload the original version on a separate branch in case it helps anyone.

**Comment by samee** [comment]
Actually, I don't need to upload it. Jack's fork of the previous code is still public. Anybody interested can try this version: https://bitbucket.org/jackdoerner/absentminded-crypto-kit

Xiao and I will try to figure out what the intention behind this change was, and why it led us astray.

**Comment by jackdoerner** [comment]
Indeed. I was unaware there were updates. Now that I know, I'll wait to merge them in until this issue is resolved. A thought: you and I are both providing libraries that provide ORAM in Obliv-C, and our codebases are mostly-identical. Someday we may want to do some sort of reconciliation between them.

**Comment by samee** [comment]
Patch applied. Will close this bug after testing a bit more thoroughly. See if it is working now.
